src/broker.ts

```diff
@@ -80,6 +82,19 @@ export class Broker {
 			}, []))
 			.catch(() => q.reject<IWhenable<any>>(`Unable to find pacts for given provider '${this.options.provider}' and tags '${this.options.tags}'`));
 	}
+
+	private _configureAuth()  {
+		if (this.options.username && this.options.password) {
+			return {
+				"user": this.options.username,
+				"password": this.options.password
+			};
+		} else if (this.options.token) {
```

Comment:
just create 2 ifs instead of having an else if.  Might be good to type this function's return object like `TokenAuth | UserPassAuth`

Reply:
No probs. I'm going to remove the class as suggested.